creator-node/src/diskManager.ts

```diff
@@ -561,3 +565,160 @@ export async function sweepSubdirectoriesInFiles(
+        try {
+          await fs.unlink(nonLegacyPath)
+        } catch (e) {
+          logger.error(
+            `_copyLegacyFiles() could not remove mismatched CID file at storageLocation=${nonLegacyPath}`
+          )
+        }
+        throw new Error('CID does not match what is expected to be')
+      }
+
+      // Optionally delete legacy file now that it was moved
+      if (deleteLegacyFiles) {
+        try {
+          await fs.unlink(nonLegacyPath)
+        } catch (e) {
+          logger.error(
+            `_copyLegacyFiles() could not remove legacy file at storageLocation=${nonLegacyPath}`
+          )
+        }
+      }
+    } catch (e: any) {
+      erroredPaths[legacyPath] = e.toString()
+    }
+  }
+
+  // TODO: Increment filesMigratedFromLegacyPath Prometheus metric (success and failure labels)
+  //       Use erroredPaths.length and copiedPaths.length
+
+  return copiedPaths
+}
+
+function _getCharsInRange(startChar: string, endChar: string): string[] {
+  // Long way for reference. TODO: Remove after confirming the shorter way isn't too confusing
+
+  // const charsInRange = []
+  // for (
+  //   let char = endChar.charCodeAt(0) + 1;
+  //   char > startChar.charCodeAt(0);
+  //   char--
+  // ) {
+  //   charsInRange.push(String.fromCharCode(char))
+  // }
+  // return charsInRange
+
+  return range(endChar.charCodeAt(0), startChar.charCodeAt(0) - 1, -1).map(
+    (charCode) => String.fromCharCode(charCode)
+  )
+}
+
+function _getCharsInRanges(...ranges: string[]): string[] {
+  const charsInRanges = []
+  for (const range of ranges) {
+    charsInRanges.push(..._getCharsInRange(range.charAt(0), range.charAt(2)))
+  }
+  return charsInRanges
+}
+
+/**
+ * 1. Finds rows in the Files table that have a legacy storagePath (/file_storage/<CID or dirCID>)
+ * 2. Copies the file to to the non-legacy path (/file_storage/files/<CID or dirCID>)
+ * 3. Updates the row in the Files table to reflect the new storagePath
+ * 4. Increments Prometheus metric `filesMigratedFromLegacyPath` to reflect the number of files moved
+ * @param logger logger to print errors to
+ * @param deleteLegacyFiles whether or not to delete the file at the legacy path after it's copied to the new path
+ */
+export async function migrateFilesWithLegacyStoragePaths(
+  logger: Logger,
+  deleteLegacyFiles: boolean
+): Promise<void> {
+  const BATCH_SIZE = 100
+  try {
+    // Paginate at each character in range [QmZ, ..., QmA, Qmz, ..., Qma, Qm9, ..., Qm0]
+    for (const char of _getCharsInRanges('a-z', 'A-Z', '0-9')) {
+      const cursor = 'Qm' + char
+
+      // Query for legacy storagePaths in the pagination range until no more results are returned
+      let legacyStoragePathsAndCids = []
+      do {
+        legacyStoragePathsAndCids =
+          await DbManager.getNonDirLegacyStoragePathsAndCids(cursor, BATCH_SIZE)
+
+        const copiedFilePaths = await _copyLegacyFiles(
+          legacyStoragePathsAndCids,
+          deleteLegacyFiles,
```

Comment:
instead of passing in `deleteLegacyFiles` here, its prob safer to do it only after `DbManager.updateLegacyPathDbRows` is confirmed successful?

Reply:
good call. moved to delete after db transaction